File: data-processing/Processor.py
from pyspark.sql import SparkSession

class Processor:
    def main():
        spark = SparkSession.builder.appName("processor").getOrCreate()

        baseURL = "s3a://gitlawbucket"
        congress = [116, 115, 114, 113]
        docuType = ["amendments", "bills"]
        house = ["hconres", "hjres", "hr", "hres", "s", "sconres", "sjres", "sres"]
        textVersion = "text-versions"

        lines = spark.read.option("multiLine", "true").option("mode", "PERMISSIVE").json("s3a://gitlawbucket/116/bills/sconres/sconres1/data.json")
        print(lines.first())
        


    if __name__ == "__main__":
        main()

File: data-processing/Processor.scala
import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.SQLContext
import scala.collection.JavaConverters._
// import com.amazonaws.services.s3.AmazonS3Client
// import com.amazonaws.services.s3.model.ListObjectsV2Request


object Processor {
    val baseURL = "s3a://gitlawbucket"
    val congress = List(116, 115, 114, 113)
    val docuType = List("amendments", "bills")
    val house = List("hconres", "hjres", "hr", "hres", "s", "sconres", "sjres", "sres")
    val textVersion = "text-versions"
    
    def main(args: Array[String]): Unit = {
        // create Spark context and SQL context using SparkSession
        val spark = SparkSession
        .builder()
        .appName("Processor")
        .getOrCreate()

        // open files
        val df = spark.read.json("s3a://gitlawbucket/116/bills/sconres/sconres1/data.json")
        // df.show
        df.first()
    }
}
